### PR TITLE
feat(auth): Add fields to client and api scope.

### PR DESCRIPTION
**Pull request by valurefugl**
https://www.notion.so/Add-requireApiScopes-to-client-db-model-adminUI-931c6637609845f38481906876fff416
https://www.notion.so/Add-grantToAuthenticatedUser-to-apiscope-db-model-adminUI-07b9e1b148a645d490db25e06091871e
https://www.notion.so/Filter-in-outgoing-delegations-5a2c0e1e55a84d279bafde9aa7f5fd85
https://www.notion.so/Tests-15a0cf55e021400db2d7a60dbec1f684

## What

Add new fields to IDS client and api scope tables.

## Why

Part of https://www.notion.so/Filtered-delegations-2836c89273f744538e200a0f8f56233f

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/auth-api-lib/src/lib/clients/dto/base/client-base.dto.ts [hunk 1]
```diff
@@ -314,4 +314,11 @@ export abstract class ClientBaseDTO {
     example: true,
   })
   readonly promptDelegations!: boolean
+
+  @IsBoolean()
+  @IsNotEmpty()
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/clients/dto/base/client-base.dto.ts:319]: `@IsNotEmpty` is unnecessary check as IsBoolean will assert a boolean value.

File: libs/auth-api-lib/src/lib/resources/dto/api-scopes.dto.ts [hunk 2]
```diff
@@ -79,6 +79,13 @@ export class ApiScopesDTO {
   })
   readonly emphasize!: boolean
 
+  @IsBoolean()
+  @IsNotEmpty()
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/resources/dto/api-scopes.dto.ts:83]: Ditto on `IsNotEmpty`

**Review by saevarma — APPROVED** [review]
Nice job 💯
Just two nice to have comments.

File: libs/auth-api-lib/src/lib/clients/dto/base/client-base.dto.ts (hunk 1 above)
Comment (valurefugl) [reply libs/auth-api-lib/src/lib/clients/dto/base/client-base.dto.ts]: Removed.

**Pull request merged**